sdk/openai/Azure.AI.OpenAI/README.md

```diff
@@ -301,26 +301,70 @@ calls, and the tool messages that resolved each of those tools -- when making a
 ChatChoice responseChoice = response.Value.Choices[0];
 if (responseChoice.FinishReason == CompletionsFinishReason.ToolCalls)
 {
-    List<ChatRequestToolMessage> toolCallResolutionMessages = new();
+    // Add the assistant message with tool calls to the conversation history
+    ChatRequestAssistantMessage toolCallHistoryMessage = new(responseChoice.Message);
+    chatCompletionsOptions.Messages.Add(toolCallHistoryMessage);
+
+    // Add a new tool message for each tool call that is resolved
     foreach (ChatCompletionsToolCall toolCall in responseChoice.Message.ToolCalls)
     {
-        toolCallResolutionMessages.Add(GetToolCallResponseMessage(toolCall));
+        chatCompletionsOptions.Messages.Add(GetToolCallResponseMessage(toolCall));
     }
 
-    // Include the ToolCall message from the assistant in the conversation history, too
-    var toolCallHistoryMessage = new ChatRequestAssistantMessage(responseChoice.Message.Content);
-    foreach (ChatCompletionsToolCall requestedToolCall in responseChoice.Message.ToolCalls)
+    // Now make a new request with all the messages thus far, including the original
+}
+```
+
+When using tool calls with streaming responses, accumulate tool call details much like you'd accumulate the other
+portions of streamed choices, in this case using the accumulated `StreamingToolCallUpdate` data to instantiate new
+tool call messages for assistant message history. Note that the model will ignore `ChoiceCount` when providing tools
+and that all streamed responses should map to a single, common choice index in the range of `[0..(ChoiceCount - 1)]`.
+
+```C# Snippet:ChatTools:StreamingChatTools
+Dictionary<int, string> toolCallIdsByIndex = new();
+Dictionary<int, string> functionNamesByIndex = new();
+Dictionary<int, StringBuilder> functionArgmentBuildersByIndex = new();
+StringBuilder contentBuilder = new();
+
+await foreach (StreamingChatCompletionsUpdate chatUpdate
+    in await client.GetChatCompletionsStreamingAsync(chatCompletionsOptions))
+{
+    if (chatUpdate.ToolCallUpdate is StreamingFunctionToolCallUpdate functionToolCallUpdate)
     {
-        toolCallHistoryMessage.ToolCalls.Add(requestedToolCall);
+        if (functionToolCallUpdate.Id != null)
+        {
+            toolCallIdsByIndex[functionToolCallUpdate.ToolCallIndex] = functionToolCallUpdate.Id;
+        }
+        if (functionToolCallUpdate.Name != null)
+        {
+            functionNamesByIndex[functionToolCallUpdate.ToolCallIndex] = functionToolCallUpdate.Name;
+        }
+        if (functionToolCallUpdate.ArgumentsUpdate != null)
+        {
+            StringBuilder argumentsBuilder
+                = functionArgmentBuildersByIndex.TryGetValue(
+                    functionToolCallUpdate.ToolCallIndex,
+                    out StringBuilder existingBuilder) ? existingBuilder : new StringBuilder();
```

Comment:
The builder created in the else branch is never stored back into the dictionary. Doesn't that mean all arguments will actually end up being lost?